src/utils/hasInterpolatingAmpersand.js

```diff
@@ -0,0 +1,23 @@
+import _ from "lodash"
+
+/**
+ * Check whether a selector has an interpolating ampersand
+ * An "interpolating ampersand" is an "&" used to interpolate within another
+ * simple selector (e.g. `&-modifier`), rather than an "&" that stands
+ * on its own as a simple selector (e.g. `& .child`)
+ *
+ * @param {string} selector
+ * @return {boolean} If `true`, the selector has an interpolating ampersand
+ */
+export default function (selector) {
+  for (let i = 0, l = selector.length; i < l; i++) {
```

Comment:
Small thing, but there is really no need to save `selector.length` into a variable. There is no perf boost from that in modern JS engines.